ADR/ADR.py

```diff
@@ -1006,34 +1030,41 @@ class ADRServer(DeviceServer):
 
     @inlineCallbacks
     def findDevices(self):
-        """
-        Finds all ADR configurations in the registry at CONFIG_PATH and returns a list of
-        (ADR_name,(),peripheralDictionary).
-        INPUTS - none
-        OUTPUT - List of (ADRName,(connectionObject,context),peripheralDict) tuples.
+        """Find all ADR configurations in the registry.
+
+        Loads configuration from the registry at CONFIG_PATH and returns a list
+        of information about ADR devices found there. The list only includes
+        ADR devices for which all required node servers are present, with the
+        node servers inferred from the names of the configured peripheral
+        devices.
+
+        Returns:
+            (list[(str, tuple, dict)]): A list of tuples, each containing then
+            name of an ADR device, along with positional and keyword args to be
+            passed to the corresponding ADRWrapper's `connect` method.
         """
         device_list = []
         reg = self.client.registry
         yield reg.cd(CONFIG_PATH)
-        resp = yield reg.dir()
-        adr_names = resp[0].aslist
-        for name in adr_names:
-            if 'defaults' not in name:
-                # all required nodes must be present to create this device
-                yield reg.cd(name)
-                devices = yield reg.dir()
-                devices = devices[1].aslist
-                missing_nodes = []
-                for dev in devices:
-                    node = yield reg.get(dev)
-                    node = node[1].split(' ')[0]
-                    if "node_" + node.lower() not in self.client.servers:
-                        missing_nodes.append(node)
-                if not missing_nodes:
-                    device_list.append((name, (self.client,)))
-                else:
-                    print "device %s missing nodes: %s" % (name, str(list(set(missing_nodes))))
-                yield reg.cd(1)
+        dirs, keys = yield reg.dir()
+        for name in dirs:  # dirs are adr names.
+            if 'defaults' in name:
```

Comment:
do we want to exclude the 'backups' folder too? i.e. `if 'defaults in name or 'backups' in name:`

Perhaps it doesn't matter because there are no keys in the backups folder, but maybe there would be in the future.